### DISCUSSION TITLE
Cannot input Modi script

### DISCUSSION THREAD

**Discussion by abhi-deshpande** [tsujan/FeatherPad#518] [Q&A] [answered]
I was editing a TeX source where I needed to input Modi script in my content, but Modi cannot be input in FeatherPad. I can only copy-paste Modi strings from another source or editor, but cannot type it directly.

I am typing Modi with my Modi (KaGaPa Phonetic) keyboard layout on MX Linux 19.4 in FeatherPad 0.16.0. (Actually, this keyboard layout is available from xkeyboard-config ≥ 2.30, but I have installed it manually.)

I think this issue is related to Qt, but not sure about it. Is it so ?

**Comment by brucehankins** [comment]
A couple of things to note here:

1) You stated you're using MXLinux. MX, regardless of DE, is a Debian based distro. The files hosted here are from Solus, an independent distro that uses a fork of the PiSi package manager and eopkg rather than apt.

2) This would be an issue with the source or one of it's dependencies such as Qt, not with the .eopkg repackaging.

3) You've already filed an issue with the correct repository, so I'd wait for a response there https://github.com/tsujan/FeatherPad/issues/516

I did a quick Google search, but couldn't find anything relevant. Sorry, but I haven't modified the source code in anyway, and I'm definitely not a programmer, so I'm not sure how to troubleshoot. I wish you luck in finding your solution.

**Comment by tsujan** [comment]
FeatherPad supports what Qt supports.

Maybe your method of typing Modi doesn't work with Qt and you need another method for it. Sorry, I have no knowledge of Modi or how it's typed in Qt but your problem isn't cause by FeatherPad.

I transfer your report to Discussions...

**Reply by abhi-deshpande** [reply]
Let me elaborate more....

1. Modi is an indic script used to write Marathi language. It's a complex script with many intersyllabic ligatures and letter variants. It was predominantly used in between 1200s to 1940s in india.
2. I am using XKB input method to input Modi in Qt.
3. After experimenting with other Qt apps like VLC and SMplayer (although they are media players, I tried to input modi in text fields in settings or URL streaming options etc.) I found that they also suffer with this issue. This issue is not seen in any GTK+ editors or apps.
4. Scribus, partially supports Modi. I can input directly in Text Frames, but cannot input in story editor or any other text field.

Now, I am sure that this is a Qt issue. Although Qt can handle complex scripts like Devanagari, why it cannot handle Modi, is still a question for me.

What do you think ? Is it a bug in Qt ?

**Reply by tsujan** [reply]
Qt may not support your Modi keyboard layout. Whether there's another way of typing it with Qt, I don't know. You may want to report it to Qt, telling them about the details of your layout.

**Reply by abhi-deshpande** [reply]
I have reported this problem to Qt. I will update this thread when they respond me. 

Thank you @tsujan  & @brucehankins for your response.

**Comment by abhi-deshpande** [comment, marked as answer]
Finally, after some discussion, Qt developers have identified this problem as a defect in basic input system, so it doesn't relate to FeatherPad. Hopefully, it will be fixed in future versions of Qt.
